dali/kernels/imgproc/jpeg/jpeg_distortion_gpu.cuh

```diff
@@ -224,24 +226,71 @@ __global__ void JpegCompressionDistortion(const SampleDesc *samples,
       ivec2 offset{x, y};
 
       auto ycbcr = rgb_to_ycbcr_subsampled<horz_subsample, vert_subsample, T>(offset, in);
-      cb[chroma_y][chroma_x] = ycbcr.cb;
-      cr[chroma_y][chroma_x] = ycbcr.cr;
+      // Shifting to [-128, 128] before the DCT.
+      cb[chroma_y][chroma_x] = ycbcr.cb - 128.0f;
+      cr[chroma_y][chroma_x] = ycbcr.cr - 128.0f;
       for (int i = 0, k = 0; i < vert_subsample+1; i++) {
         for (int j = 0; j < horz_subsample+1; j++, k++) {
-          luma[luma_y + i][luma_x + j] = ycbcr.luma[k];
+          luma[luma_y + i][luma_x + j] = ycbcr.luma[k] - 128.0f;
         }
       }
 
       __syncthreads();
 
-      // TODO(janton): DCT + quantization + inv DCT
+      static constexpr int col_stride = 1;
+      static constexpr int row_stride = 9;
+
+      if (chroma_x <= vert_subsample) {  // 1 or 2 rows depending on vertical subsampling
+        dct_fwd_8x8_1d<col_stride>(&luma[luma_y + chroma_x][0]);
+      }
+
+      if (chroma_x == 0) {  // once per row
+        dct_fwd_8x8_1d<col_stride>(&cb[chroma_y][0]);
```

Comment:
TODO: modify so that the 4 rows (up two rows for luma and one for cb and cr each) are processed in parallel for 4 different threads.